### ISSUE TITLE
Projects

### ISSUE THREAD

**Issue by matthewpereira** [concourse/concourse#5683]
### TLDR

* A Concourse project is a namespace for resources, tasks, and pipelines. Projects are bootstrapped from a single resource definition and a build plan which executes when new versions of the resource are found. Projects bridge the gap between small-scale and large-scale CI workflows, helping to smoothen Concourse's learning curve.

### Problem

* Concourse is built with a 'teams' abstraction baked-in, where in reality many Concourse installations function with a single team or single user, and don't require this multi-tenant organizational focus.
* Concourse has a steep learning curve compared to other CI/CD tools like Travis or Circle, which allow single-build workflows without needing pipelines at all
* It can be difficult to fully automate a medium-to-large scale usecase via the set_pipeline step ability without a per-project abstraction layer
* Concourse requires passing task configs as a resource throughout a pipeline in order to ensure pipelines and tasks are in sync 

### Proposed Solution

- Introduce Projects as a method to allow a user to bootstrap and automate their CI/CD stack by configuring a single resourced which contains their project's config:

```sh
fly set-project \
  --project booklit \
  --type git \
  --source uri=https://github.com/vito/booklit \
  --source branch=projects-mock \
  --path ci
```
- The resource must contain a project config as `project.yml` under the configured `--path`.
- This path also contains resource, resource type, task, and pipeline definitions for the project. A full project config directory may look something like this:

```sh
ci/project.yml
ci/resource-types/github-status.yml
ci/resources/booklit.yml
ci/resources/booklit-status.yml
ci/tasks/test.yml
ci/tasks/build.yml
ci/pipelines/booklit.yml
```

### Outcomes

* Easily template and automate the setup of many workflows in a prescribed, compliant manner
* Simplify the product for those who don't have a need for its more advanced workflows, without sacrificing any of that advanced functionality in the process
* A project can be defined with project-wide resources which allow you to remove duplicate definitions across your pipleines and support cross-pipeline passed constraints
* Projects also define project-wide tasks, which remove the need to thread a resource through all your jobs just to have the task configs to execute, and finally gives meaning to task names (the x in task: x).
* Other [v10](https://github.com/concourse/concourse/labels/v10) features are made all the more powerful with the addition of this workflow

### Current Status

- Slated to begin engineering efforts in Q1 2022.

### References

See [RFC](https://github.com/vito/rfcs/blob/projects/032-projects/proposal.md)
See [RFC discussion](https://github.com/concourse/rfcs/pull/32)

**Comment by taylorsilva** [comment]
# Issue Bankruptcy

This issue is being closed as part of 2025's **Issue Bankruptcy** declaration. [See here for details](https://blog.concourse-ci.org/posts/2025-01-06-concourse-2024-review/)

If you think this issue still has merit then feel free to open a new issue and reference this one.
